src/sql/workbench/services/notebook/browser/models/cell.ts

```diff
@@ -523,6 +526,25 @@ export class CellModel extends Disposable implements ICellModel {
 		return this._outputs;
 	}
 
+	public updateOutputData(batchId: number, id: number, data: any) {
+		for (let i = 0; i < this._outputs.length; i++) {
+			if (this._outputs[i].output_type === 'execute_result'
+				&& (<nb.IExecuteResult>this._outputs[i]).batchId === batchId
+				&& (<nb.IExecuteResult>this._outputs[i]).id === id) {
+				(<nb.IExecuteResult>this._outputs[i]).data = data;
+				break;
+			}
+		}
+	}
+
+	public get gridDataConversionComplete(): Promise<void[]> {
```

Comment:
void[] seems like an odd return type, I would just make it void and change the method to `Promise.all(this._gridDataConversionComplete).then()`